roxie/ccd/ccdserver.cpp

```diff
@@ -5210,11 +5219,15 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
             else
             {
                 unsigned timeNow = msTick();
-                if (!anyActivity && !localAgent && (timeNow-lastActivity >= timeout))
+                if (acknowledgeAllRequests)
+                {
+                    retryPending(checkInterval);
+                }
+                else if (!anyActivity && !localAgent && (timeNow-lastActivity >= checkInterval))
                 {
                     lastActivity = timeNow;
-                    activity.queryLogCtx().CTXLOG("Input has stalled for %u ms - retry required?", timeout);
-                    retryPending();
+                    activity.queryLogCtx().CTXLOG("Input has stalled for %u ms - retry required?", checkInterval);
+                    retryPending(0);
```

Comment:
does this mean that retries will be send 10x quicker than before?

Is the following the difference?
if acknowledgeAllRequests is true:
  if there has been no acknowledgement for a packet within the checkInterval then it will be re-sent.

if acknowledgeAllRequests is false:
  if there is no activity in the checkInterval then all packets will request to be resent.

If so it is probably worth documenting somewhere.

What is the difference if acknowledgeAll is set

Reply:
Yes 10x quicker retries.   I think as it is now - if the query timeout is less than the HIGH/LOW timeout then no retries at all.
And we still have the bug where queries copied always get LOW priority.

Reply:
The 10x faster could be viewed as a separate though related fix - it's what the changes in HPCC-25452 were supposed to do, but failed...

acknowledgeAllRequests enables the new behaviour of resending any request that has not seen an ack, rather than requiring ALL requests to stall for the timeout period before any get resent. It's not an option I intend to document though - the option is there to enable us to test and observe the difference (and to disable should it cause unanticipated problems) rather than for end-users to decide which they prefer...